[oneTBB] Heterogeneous overloads for concurrent_hash_map (#356)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/containers/concurrent_hash_map_cls/modifiers.rst

        bool insert( accessor& result, const key_type& key );

    If the ``result`` accessor is not empty, releases the ``result`` and
    attempts to insert the value constructed from ``key, mapped_type()`` into the container.

    Sets the ``result`` to provide access to the inserted element or to the element with equal key
    which was already presented in the container.

    **Requirements**:

>     * the ``value_type`` type must meet the ``EmplaceConstructible`` requirements the from [container.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
    * the ``value_type`` type must meet the ``EmplaceConstructible`` requirements described in the [container.requirements] section of the ISO C++ Standard.
```

>     * the ``mapped_type`` type must meet the ``DefaultConstructible`` requirements from the [defaultconstructible] ISO C++ Standard section.

Reviewer: ```suggestion
    * the ``mapped_type`` type must meet the ``DefaultConstructible`` requirements described in the [defaultconstructible] section of the ISO C++ Standard.
```

    **Returns**: ``true`` if an element is inserted; ``false`` otherwise.

--------------------------

    .. code:: cpp

        template <typename K>
        bool insert( const_accessor& result, const K& key );

        template <typename K>
        bool insert( accessor& result, const K& key );

>     If the ``result`` accessor is not empty, releases the ``result`` and

Reviewer: Better to remove indent here and later on, otherwise, it'll be built as part of the code block

Author: It seems like we use the same indent in such blocks and it renders correct

    **Returns**: ``true`` if an element was inserted; ``false``, otherwise.

--------------------------

    .. code:: cpp

        template <typename K>
        bool insert( const_accessor& result, const K& key );

        template <typename K>
        bool insert( accessor& result, const K& key );

    If the accessor ``result`` is not empty, releases the ``result`` and
    attempts to insert the value constructed from ``key, mapped_type()`` into the container.

    Sets the ``result`` to provide access to the inserted element or to the element with the key

>     which compares equivalent with ``key``, that was already presented in the container.

Reviewer: ```suggestion
    that compares equivalent with ``key``, which was already presented in the container.
```

        bool insert( accessor& result, const value_type& value );

    If the ``result`` accessor is not empty, releases the ``result`` and
    attempts to insert the value ``value`` into the container.

>     Sets the ``result`` to provide access to the inserted element or to the element with equal key

Reviewer: ```suggestion
    Sets the ``result`` to provide access to the inserted element or to the element with equal key,
```

    which was already presented in the container.

>     **Requirements**: the ``value_type`` type must meet the ``CopyInsertable`` requirements from the

Reviewer: ```suggestion
    **Requirements**: the ``value_type`` type must meet the ``CopyInsertable`` requirements described in the
```

>     [container.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
    [container.requirements] section of the ISO C++ Standard.
```

    Attempts to insert the value ``value`` into the container.

>     **Requirements**: the ``value_type`` type must meet the ``CopyInsertable`` requirements from the

Reviewer: ```suggestion
    **Requirements**: the ``value_type`` type must meet the ``CopyInsertable`` requirements described in the
```

>     [container.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
    [container.requirements] section of the ISO C++ Standard.
```

        bool insert( accessor& result, value_type&& value );

    If the ``result`` accessor is not empty, releases the ``result`` and
    attempts to insert the value ``value`` into the container using move semantics.

>     Sets the ``result`` to provide access to the inserted element or to the element with equal key

Reviewer: ```suggestion
    Sets the ``result`` to provide access to the inserted element or to the element with equal key,
```

    which was already presented in the container.

    ``value`` is left in a valid, but unspecified state.

>     **Requirements**: the ``value_type`` type must meet the ``MoveInsertable`` requirements from the

Reviewer: ```suggestion
    **Requirements**: the ``value_type`` type must meet the ``MoveInsertable`` requirements described in the
```

>     [container.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
    [container.requirements] section of the ISO C++ Standard.
```

    Attempts to insert the value ``value`` into the container using move semantics.

>     **Requirements**: the ``value_type`` type must meet the ``MoveInsertable`` requirements from the

Reviewer: ```suggestion
    **Requirements**: the ``value_type`` type must meet the ``MoveInsertable`` requirements described in the
```

>     [container.requirements] ISO C++ Standard section.

Reviewer: ```suggestion
    [container.requirements] section of the ISO C++ Standard.
```

    If the interval ``[first, last)`` contains multiple elements with equal keys,
    it is unspecified which element should be inserted.

>     **Requirements**: the ``InputIterator`` type must meet the requirements of `InputIterator`

Reviewer: ```suggestion
    **Requirements**: the ``InputIterator`` type must meet the ``InputIterator`` requirements described in the
```

>     from the ``[input.iterators]`` ISO C++ Standard section.

Reviewer: ```suggestion
    ``[input.iterators]`` section of the ISO C++ Standard.
```

        template <typename... Args>
        bool emplace( accessor& result, Args&&... args );

    If the ``result`` accessor is not empty, releases the ``result`` and
    attempts to insert an element constructed in-place from ``args`` into the container.

>     Sets the ``result`` to provide access to the inserted element or to the element with equal key

Reviewer: ```suggestion
    Sets the ``result`` to provide access to the inserted element or to the element with equal key,
```

    If an element with the key equivalent to  ``key`` exists, removes it from the container.

>     **Returns**: ``true`` if an element is removed; ``false``, otherwise.

Reviewer: ```suggestion
    **Returns**: ``true`` if an element is removed; ``false`` otherwise.
```

--------------------------

    .. code:: cpp

        template <typename K>
        bool erase( const K& key );

>     If an element with the key that compares equivalent to the value ``key`` exists, removes it from the container.

Reviewer: Same with indent here

    This overload only participates in the overload resolution if qualified-id
    ``hash_compare_type::is_transparent`` is valid and denotes a type.

>     **Returns**: ``true`` if an element is removed; ``false``, otherwise.

Reviewer: ```suggestion
    **Returns**: ``true`` if an element is removed; ``false`` otherwise.
```

    **Requirements**: ``item_accessor`` should not be empty.

    **Returns**: ``true`` if an element is removed by the current thread; ``false``

>     if it was removed by another thread.

Reviewer: ```suggestion
    if it is removed by another thread.
```
